lib/matplotlib/quiver.py

```diff
@@ -626,19 +626,19 @@ def _angles_lengths(self, U, V, eps=1):
 
     def _make_verts(self, U, V, angles):
         uv = (U + V * 1j)
-        str_angles = isinstance(angles, six.string_types)
-        if str_angles and (angles == 'xy' and self.scale_units == 'xy'):
+        str_angles = angles if isinstance(angles, six.string_types) else ''
```

Comment:
Build mpl on latest commit 1144e8c0b89434888a15b1148e1aea21db9878d4 on branch master

Run
```python
_, ax = plt.subplots(subplot_kw=dict(aspect='equal'))
ax.quiver(0, 0, 1, 1, angles='xy', scale_units='xy', scale=1)
ax.set_xlim(0, 1.1); ax.set_ylim(0, 1.1); ax.grid()
```
[before.pdf](https://github.com/matplotlib/matplotlib/files/1033118/before.pdf)
[after.pdf](https://github.com/matplotlib/matplotlib/files/1033117/after.pdf)

this block l. 668 is wrong:
```python
        if str_angles and (angles == 'xy'):
            theta = angles
        elif str_angles and (angles == 'uv'):
            theta = np.angle(uv)
```
since variable angles is overridden by `angles, lengths = self._angles_lengths(U, V, eps=1)`